pkg/network/tracer/tracer.go

```diff
@@ -522,6 +522,9 @@ func (t *Tracer) getConnections(activeBuffer *network.ConnectionBuffer) (latestU
 		t.addProcessInfo(&activeConnections[i])
 	}
 
+	// get rid of stale process entries in the cache
+	t.processCache.Trim()
```

Comment:
should we do this independently of the `getConnections` check?  It seems like it is being populated asynchronously already